### ISSUE TITLE
Langserver panics after attempting to autocomplete new define

### ISSUE THREAD

**Issue by sabiram** [SpaceManiac/SpacemanDMM#57]
Langserver panics and spams error (#55) upon attempting to autocomplete a **new** define. After the panic, autocomplete now recognises the define and works properly.

```dm

#define MATERIAL_MAHOGANY "mahogany"

/turf/simulated/wall/wood/New(var/newloc)
	..(newloc,MATERIAL_WOOD)

/turf/simulated/wall/mahogany/New(var/newloc)
	..(newloc,MATERIAL_MAHOGANY)
```

Complete steps I take to reproduce:
1. Create new define (MATERIAL_MAHOGANY)
1. Copy-paste existing obj (in this case, `.../wall/wood/New`)
1. Double-click MATERIAL_WOOD to select text
1. Type M to begin autocompleting MATERIAL
1. Panic

**Comment by SpaceManiac** [comment]
Finally managed to extract a usable trace. The completion is looking at the updated version of the file where there's a second copy of the same proc, and trying to reference the second definition of the proc from the outdated object tree, which contains only the first definition. Thus, OOB.

**Comment by SpaceManiac** [comment]
Fixed two places that cause this in 9502aef1aae26f3e101cb417f1d2dd6d10d4ece1.
